aws/src/main/java/org/apache/iceberg/aws/glue/GlueToIcebergConverter.java

```diff
@@ -45,7 +45,7 @@ static TableIdentifier toTableId(Table table) {
    */
   static void validateTable(Table table, String fullName) {
     String tableType = table.parameters().get(BaseMetastoreTableOperations.TABLE_TYPE_PROP);
-    ValidationException.check(
+    NoSuchIcebergTableException.check(
```

Comment:
From the perspective of this util method, it does not make much sense to throw `NoSuchIcebergTableException`. I would propose just do a try-catch in `doRefresh` and throw the right exception there and add a few lines of documentation for why.